emu/src/cpu/arm7tdmi.rs

```diff
@@ -611,9 +611,12 @@ impl Arm7tdmi {
 
     pub fn pc_relative_load(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
         let rd = op_code.get_bits(8..=10);
-        let value = op_code.get_bits(0..=7);
+        let address = op_code.get_bits(0..=7) as usize;
+        let value = self.memory.lock().unwrap().read_word(address);
+        let mut pc = self.registers.program_counter() as u32;
+        pc.set_bit_off(0);
         self.registers
-            .set_register_at(rd.try_into().unwrap(), value as u32);
+            .set_register_at(rd.try_into().unwrap(), (pc + 4 + value) << 2);
```

Comment:
I think the address should be `pc + 4 + (immediate << 2)` because we should read an address which is relative to the PC. What do you think?

Reply:
Yes seems you're right

Reply:
Mhh I still think there's something off. What I understood from the documentation is that we construct the address by doing `pc + 4 + (value << 2)` and then we read the memory at that address. Right now we're reading at `imm8` address.

Reply:
Something like:

```rust
let address = pc + 4 + (value << 2); // where value is the immediate in the opcode
let value = self.memory.lock().unwrap().read_word(address);
self.registers.set_register_at(rd, address);
```

Reply:
Ok now looks correct :)